File: 04-Secuencial/02-Donacion/01-Python/Donacion.py
"""
    Programa para la iniciativa Amigo Social que permita generar 
    el documento de recibido que sirva como soporte contable de 
    la donación.
    Autor: Julián Esteban Gutiérrez Posada
    Fecha: Febrero 2025
    Licencia: GNU GPL v3
"""

from modulo.util import mostrar_mensaje, ingresar_texto

def main():
    nombre_casa = ingresar_texto("Nombre casa adulto mayor: ")
    cantidad_recolectada = ingresar_texto("Cantidad recolectada: ")
    recibo = generar_recibo(nombre_casa, cantidad_recolectada)
    mostrar_mensaje(recibo)

def ingresar_entero(pregunta):
    return int(ingresar_texto(pregunta))

def generar_recibo(nombre_casa, cantidad_recolectada):
    return (
        f"\nLa iniciativa Amigo Social tiene el gusto de entregar"
        f"\nuna donación de ${cantidad_recolectada} pesos colombianos"
        f"\na la casa del adulto mayor {nombre_casa}."
        f"\n\n_______________________"
        f"\nFirma representante legal"
    )

main()

File: 04-Secuencial/02-Donacion/02-Go/Donacion.go
/*
   Programa para la iniciativa Amigo Social que permita generar
   el documento de recibido que sirva como soporte contable de
   la donación.
   Autor: Julián Esteban Gutiérrez Posada
   Fecha: Febrero 2025
   Licencia: GNU GPL v3
*/

package main

import (
	"destacado/modulo"
	"fmt"
	"strconv"
)

func main() {
	nombreCasa := modulo.IngresarTexto("Nombre casa adulto mayor: ")
	cantidadRecolectada := ingresarEntero("Cantidad recolectada: ")
	reciboDonacion := generarRecibo(nombreCasa, cantidadRecolectada)
	modulo.MostrarMensaje(reciboDonacion)
}

func ingresarEntero(pregunta string) int {
	entero, _ := strconv.Atoi(modulo.IngresarTexto(pregunta))
	return entero
}

func generarRecibo(nombreCasa string,
	cantidadRecolectada int) string {
	return fmt.Sprintf(
		"\nLa iniciativa Amigo Social tiene el gusto de entregar"+
			"\nuna donación de $%d pesos colombianos"+
			"\na la casa del adulto mayor %s.", cantidadRecolectada, nombreCasa+
			"\n\n_______________________"+
			"\nFirma representante legal")
}
